core/sqf/src/seatrans/tm/hbasetmlib2/src/main/java/org/trafodion/dtm/HBaseTxClient.java

```diff
@@ -1062,6 +1062,14 @@ public void run() {
                                               tmID + " regionBytes: [" + regionBytes + "].  Deleting zookeeper region entry. \n exception: " + de);
                                       zookeeper.deleteRegionEntry(regionEntry);
                                 }// DeserializationException
+                                catch (IOException ioe) {
+                                    // It's possible we received a spurious entry from a stale region.
+                                       // It's safe to delete the zookeeper entry and wait for any new posts
+                                       LOG.warn("TRAF RCOV THREAD:IOException calling txnManager.recoveryRequest. " + "TM: " +
+                                               tmID + " regionBytes: [" + regionBytes
+                                               + "].  Deleting zookeeper region entry. exception: " + ioe);
+                                       zookeeper.deleteRegionEntry(regionEntry);
+                                 }// IOException
```

Comment:
Retrying the coprocessor call inline is not necessary.  The only thing required is deleting the ZKNode that caused the exception.  If a TableNotFoundException was the the issue, we should not retry, because the table doesn't exist and it won't ever exist.  We should just delete the entry and move on.  It was the fact that we were leaving the the ZKNode in place that caused the incessant retries and exceptions that brought the TM down.

If the region really does require recovery, it will post a new ZKNode and that will get picked up by the TM.  That is the intended design of the recovery mechanism.